### PR TITLE
Make SQLServer join work

### PR DESCRIPTION
**Pull request by AdRiley**
### Pull Request Description

Enable Join feature on SQLServer. Case sensitivity again makes some different results...

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [x] Unit tests have been written where possible.
- [x] If meaningful changes were made to logic or tests affecting Enso Cloud integration in the libraries, 
      or the Snowflake database integration, a run of the [Extra Tests](https://github.com/enso-org/enso/actions/workflows/extra-nightly-tests.yml) has been scheduled.
  - If applicable, it is suggested to paste a link to a successful run of the Extra Tests.
  - https://github.com/enso-org/enso/actions/runs/11728362663

### CODE REVIEW

File: test/Table_Tests/src/Common_Table_Operations/Join/Join_Spec.enso [hunk 1]
```diff
@@ -146,9 +146,15 @@ add_join_specs suite_builder setup =
...
-            r1 . at "Z" . to_vector . should_equal [2]
+            case setup.flagged ..Case_Sensitive_Text_Comparison of
+                True ->
+                    r1 . at "X" . to_vector . should_equal ["a"]
+                    r1 . at "Y" . to_vector . should_equal [1]
+                    r1 . at "Z" . to_vector . should_equal [2]
+                False ->
+                    r1 . at "X" . to_vector . should_equal ["a", "a", "B"]
+                    r1 . at "Y" . to_vector . should_equal [1, 1, 2]
+                    r1 . at "Z" . to_vector . should_equal [1, 2, 3]
```
Comment (radeusgd) [inline test/Table_Tests/src/Common_Table_Operations/Join/Join_Spec.enso:157]: with multiple results, let's add `|> materialize |> _.sort ["Z"]` to ensure the ordering is consistent. Adding `materialize` will also mean we execute the query only once, not 3 times (on every `to_vector` call - which is better for high-latency backends such as Snowflake (I'm surprised I didn't notice we don't materialize here when working on Snowflake tests).

**Review by radeusgd — APPROVED** [review]
Overall changes look good.

Does it mean that on SQL Server the `Equals` and `Equals_Ignore_Case` join conditions are currently equivalent?

I guess it makes sense to make the `Equals` rely on the default string comparison (to allow it to be efficient and use whatever indexes are there in the DB) - so if SQL Server is case insensitive - let it be case insensitive.

**But** I feel like we should still have some way to force the comparison to be case sensitive. What do you think? Ofc I'd suggest to create a ticket and do it as separate task.

---

Maybe we could add a `force_case_sensitive:Boolean=False` to the `Equals` condition? Backends that are sensitive by default will ignore it, but backends that are case insensitive would see if they can keep using case insensitive comparison or if they need to force sensitivity by some means. Then, if `force_case_sensitive=True`, and a backend does not support case sensitive comparisons at all, we could report an error/warning.

> Or instead of adding a parameter that may be confusing, we could add a new Join Condition: `Equals_Case_Sensitive`. In most backends `Equals_Case_Sensitive` and `Equals` would be the same, but it would allow specifically selecting the case sensitivity vs relying on the default behaviour. This would probably most closely resemble our `Case_Sensitivity` type which is a tri-state: `Sensitive`, `Default` and `Insensitive`.

All in all, what I don't like is the silent automatic switchover from a case sensitive to insensitive comparison. While OK for a default behaviour, I think that I'd want to have an option to explicitly ask for case sensitive comparison and either get it or get an error saying that it is unsupported on this backend.

File: test/Table_Tests/src/Common_Table_Operations/Join/Join_Spec.enso (hunk 1 above)
Comment (AdRiley) [reply test/Table_Tests/src/Common_Table_Operations/Join/Join_Spec.enso:157]: Done

**Comment by AdRiley** [comment]
> Equals_Case_Sensitive

Yes. I think this is what we should do. As you say mirrors the Case_Sensitivity tri-state

**Review by GregoryTravis — APPROVED** [review]

**Pull request merged**